app/org/sagebionetworks/bridge/dao/UploadSchemaDao.java

```diff
@@ -0,0 +1,11 @@
+package org.sagebionetworks.bridge.dao;
+
+import org.sagebionetworks.bridge.models.upload.UploadSchema;
+
+public interface UploadSchemaDao {
+    UploadSchema createOrUpdateUploadSchema(String studyId, String schemaId, UploadSchema uploadSchema);
```

Comment:
When creating a new schema, it looks like the schema id is supplied by the user? As it is part of the hash key, I would suggest we control the creation of the schema id instead.  The user is responsible for supplying a good name.  Otherwise, the id seems duplicate of the schema name.  Alternatively, we can get rid of the id and use (name, revision) to reference a schema.